src/range_proof.rs

```diff
@@ -54,4 +58,14 @@ pub trait RangeProofService {
     /// Return the maximum range of the range proof as a power of 2. i.e. if the maximum range is 2^64, this function
     /// returns 64.
     fn range(&self) -> usize;
+
+    /// Construct a rangeproof with the ability to rewind it. Requires two rewind keys and a 19-byte message to be included in the range proof.
+    /// The proof can contain 23 bytes but 4 bytes are used to confirm that a rewind was performed correctly
+    fn construct_proof_with_rewind_key(&self, key: &Self::K, value: u64, rewind_key: &Self::K, rewind_blinding_key: &Self::K, proof_message: &[u8; 19]) -> Result<Self::P, RangeProofError>;
+
+    /// Rewind a rewindable range proof to reveal the committed value and the 19 byte proof message
+    fn rewind_proof_value_only(&self, proof: &Self::P, commitment: &HomomorphicCommitment<Self::PK>, rewind_public_key: &Self::PK, rewind_blinding_public_key: &Self::PK) -> Result<(u64, [u8; 19]), RangeProofError>;
+
+    /// Fully rewind a rewindable range proof to reveal the committed value, blinding factor and the 19 byte proof message.
+    fn rewind_proof_commitment_data(&self, proof: &Self::P, commitment: &HomomorphicCommitment<Self::PK>, rewind_key: &Self::K, rewind_blinding_key: &Self::K) -> Result<(u64, Self::K, [u8; 19]), RangeProofError>;
```

Comment:
I'd consider a wrapper struct or trait here. Tuples in public APIs are bad for readability.